### DISCUSSION TITLE
Use react-query but no react ...

### DISCUSSION THREAD

**Discussion by zevenbergm** [TanStack/query#790] [General]
Hi All, 

Thanks for this awesome project on GitHub! I love the react-query way of working with the server state; however, I'm working on a project without React... It should be wonderful if I could use the core functionalities only? I've played around with this idea and it seems to be working: https://codesandbox.io/s/magical-bush-gbsk7?file=/src/index.js ... or do I miss something important? The only thing which is not working is the refetch interval. If possible, I would like to avoid the dependy on React. How can this be accomplished? Maybe it's an idea to substract the core parts in another project and NPM library? :)

Thanks,
Martin

**Comment by mehdi-cit** [comment]
Any luck pursuing this @zevenbergm ?
I'm also very interested in using react-query but without React.js (in a vanila js project for my case)
Cheers~

**Reply by zevenbergm** [reply]
@mehdi-cit: It seems to work; however, there's a development dependency on react. I prefer to have the core in a seperate npm package. Besides that, I need a confirmation from the team if the core part is correctly implemented.

@tannerlinsley: Can we use react-query as follows: https://codesandbox.io/s/magical-bush-gbsk7?file=/src/index.js? Or do you forsee any glitches ...

**Reply by stringparser** [reply]
That doesn't look too bad :))

**Comment by stringparser** [comment]
Yup is a shame this is not possible right now with vanilla js/ts only. I'm currently working on backend and would love to have this to make queries without having to re-do the wheel as you could say 😆

**Reply by TkDodo** [reply]
it's 2023. Use `@tanstack/query-core` rather than `@tanstack/react-query`.

**Reply by stringparser** [reply]
Ah didn't know thanks
